plugins/widget/plugin.js

```diff
@@ -1066,17 +1064,11 @@
 		 * @returns {Boolean} Whether the style is applied to this widget.
 		 */
 		checkStyleActive: function( style ) {
-			var classes = getStyleClasses( style ),
-				cl;
-
-			if ( !classes )
-				return false;
+			style = CKEDITOR.tools.copy( style );
+			style.element = this.element.getName();
+			style._.definition.ignoreReadonly = true;
 
-			while ( ( cl = classes.pop() ) ) {
-				if ( !this.hasClass( cl ) )
-					return false;
-			}
-			return true;
+			return CKEDITOR.style.prototype.checkElementMatch.call( style, this.element );
```

Comment:
I'm using `style.prototype` because using it like that would refer to `style.customHandler.widget.checkElementMatch` instead. But after reworking my PR I won't call it.